### PR TITLE
networkmanager: use modify_system=true

### PR DESCRIPTION
**Pull request by worldofpeace**
This does the following:
Allow users to modify system connections [0].

Technically what it does is change the NM_MODIFY_SYSTEM_POLICY
to "yes" instead of "auth_admin_keep" which would require a password prompt.
That is inacted in the following polkit policy [1].

This allows us to drop a permissive looking polkit rule, and we don't
have to instruct users to add themselves to the "networkmanager" group.

All relevant documentation is updated.

[0]: https://gitlab.freedesktop.org/NetworkManager/NetworkManager/blob/1.20.4/meson_options.txt#L17
[1]: https://gitlab.freedesktop.org/NetworkManager/NetworkManager/blob/1.20.4/data/org.freedesktop.NetworkManager.policy.in.in#L115

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jtojnar** [comment]
I am not sure this is something we want. Won't then every user including the system ones be able to modify network configuration?

**Comment by worldofpeace** [comment]
It appears the rule applied to modemanager as well
```
 polkit.addRule(function(action, subject) {
      if (
        subject.isInGroup("networkmanager")
        && (action.id.indexOf("org.freedesktop.NetworkManager.") == 0
            || action.id.indexOf("org.freedesktop.ModemManager")  == 0
        ))
          { return polkit.Result.YES; }
    });
```

so I need to see if there's any build config for that as well.

**Comment by worldofpeace** [comment]
Ok, I believe this is ready now. Probably needs a tester.
(no one volunteers it'll probably be me, though that does make the SPOF me :rofl: )

File: pkgs/tools/networking/modem-manager/default.nix [hunk 1]
```diff
@@ -24,6 +24,7 @@ stdenv.mkDerivation rec {
     "--localstatedir=/var"
     "--with-systemd-suspend-resume"
     "--with-systemd-journal"
+    "--with-polkit=permissive"
```
Comment (jtojnar) [inline pkgs/tools/networking/modem-manager/default.nix]: I am still not sure this is best practice. Though Fedora disables it completely: https://src.fedoraproject.org/rpms/ModemManager/blob/64f08c48384af5be55b8f9d668ca03613ced3780/f/ModemManager.spec#_106

**Comment by worldofpeace** [comment]
Yeah @jtojnar, I'm not sure if there's a best practice here.
Meaning, giving permissive functionality to users of a particular group `networkmanager` or just permit system users.
(and I can't remember from my polkit research if that was just wheel users)

File: pkgs/tools/networking/modem-manager/default.nix (hunk 1 above)
Comment (worldofpeace) [reply pkgs/tools/networking/modem-manager/default.nix]: I thought permissive made the most sense with what this PR does.
Other distros that did this included Arch Linux and Void Linux.

**Comment by worldofpeace** [comment]
Not stale, just needs a decision.

**Pull request closed without merging**